.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,100 @@
+name: Build and test
+
+on:
+  push:
+      branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '0 0 * * 0'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Compile
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: wrapper
+          arguments: compileTestKotlin
+      - name: Build
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build
+      - name: Upload test reports
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: test-reports
+          path: ./build/reports/tests
+      - name: Upload test results
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: test-results
+          path: ./build/test-results
+  build-check:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - run: echo "All build jobs successful."
+  test-integration:
+    runs-on: ubuntu-latest
+    needs: build-check
+    env:
+      JIRA_LICENSE: '${{ secrets.JIRA_LICENSE }}'
+      TERM: dumb
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Prepare Jira license
+        run: echo $JIRA_LICENSE > jira-license.txt
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          aws-region: eu-west-1
+      - name: Generate workflow URL
+        run: echo "WORKFLOW_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" >> $GITHUB_ENV
+      - name: Test integration
+        uses: gradle/gradle-build-action@v2
+        continue-on-error: true # expect it to sometimes flake
+        # timeout-minutes: 70
```

Comment:
This timeout is copied from the CircleCi - I'm not sure if we want to keep it. Wdyt?

Reply:
Having timeouts is always good. We could run it a few times, see how long it takes, add some buffer and put it here

Reply:
In CircleCI it was `no_output_timeout: 70m`, so "how long can it stay silent". It was not a timeout for the whole build.
OTOH [a recent `testIntegration` CI step](https://app.circleci.com/pipelines/github/atlassian/jira-hardware-exploration/385/workflows/560e7d6d-2c13-43b7-a36c-cd5df90557b7/jobs/475) took 11 minutes.
Let's see if it hangs and then add timeout as a solution to that problem.

Reply:
Yep, I checked if GHA has a counterpart for `no_output_timeout` but unfortunately they don't thus I was thinking about using just a regular timeout, but tbh I have no idea how long those tests can take to complete in the worst-case scenario. So yep, after some time we can decide if there's a neet to introduce a timeout.